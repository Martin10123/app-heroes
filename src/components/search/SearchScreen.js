import React, { useMemo } from "react";
import queryString from "query-string";
import HeroCard from "../heroes/HeroCard";
import { useForm } from "../../Hooks/useForm";
import { useLocation, useNavigate } from "react-router-dom";
import { getHeroesByName } from "../../selectors/getHeroesByName";

const SearchScreen = () => {
  const location = useLocation();
  const navigate = useNavigate();
  const { q = "" } = queryString.parse(location.search);

  const [values, handleInputChange] = useForm({ name: q });

  const { name } = values;

  const heroesFilters = useMemo(() => getHeroesByName(q), [q]);

  const handleSearch = (e) => {
    e.preventDefault();
    navigate(`?q=${name}`);
  };

  return (
    <div>
      <h1>Search Screen</h1>
      <hr />

      <div className="row">
        <div className="col-5">
          <h4> Search Form </h4>
          <hr />

          <form onSubmit={handleSearch}>
            <input
              type="text"
              name="name"
              placeholder="Find your hero"
              className="form-control"
              value={name}
              onChange={handleInputChange}
            />

            <button
              type="submit"
              className="btn m-1 btn-clock btn-outline-primary"
            >
              Search...
            </button>
          </form>
        </div>
        <div className="col-7">
          <h4>Results</h4>
          <hr />

          {heroesFilters.map((hero) => (
            <HeroCard key={hero.id} {...hero} />
          ))}
        </div>
      </div>
    </div>
  );
};

export default SearchScreen;
