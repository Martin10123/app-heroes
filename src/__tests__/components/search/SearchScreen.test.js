import { mount } from "enzyme";
import { MemoryRouter } from "react-router-dom";
import SearchScreen from "../../../components/search/SearchScreen";

describe("Pruebas en el componente <SearchScreen/>", () => {
  test("Debe de mostrar el componente correctamente", () => {
    const wrapper = mount(
      <MemoryRouter initialEntries={["/search"]}>
        <SearchScreen />
      </MemoryRouter>
    );

    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find("h4").at(1).text().trim()).toBe("Results");
  });
});
