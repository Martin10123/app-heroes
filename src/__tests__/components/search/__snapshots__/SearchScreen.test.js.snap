// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el componente <SearchScreen/> Debe de mostrar el componente correctamente 1`] = `
<div>
  <h1>
    Search Screen
  </h1>
  <hr />
  <div
    className="row"
  >
    <div
      className="col-5"
    >
      <h4>
         Search Form 
      </h4>
      <hr />
      <form
        onSubmit={[Function]}
      >
        <input
          className="form-control"
          name="name"
          onChange={[Function]}
          placeholder="Find your hero"
          type="text"
          value=""
        />
        <button
          className="btn m-1 btn-clock btn-outline-primary"
          type="submit"
        >
          Search...
        </button>
      </form>
    </div>
    <div
      className="col-7"
    >
      <h4>
        Results
      </h4>
      <hr />
    </div>
  </div>
</div>
`;
